### PR TITLE
JDM Update and Linking Rules

### PR DESCRIPTION
**Pull request by timwekkenbc**
- [x] Update JDM editor to latest version
- [x] Added custom component and functionality for linking rules
- [x] Updated how simulator works to comply with JDM changes

### CODE REVIEW

File: app/components/SimulationViewer/SimulationViewer.tsx [hunk 1]
```diff
@@ -40,33 +42,65 @@ export default function SimulationViewer({
   };
 
   const ruleMapInputs = createRuleMap(rulemap.inputs);
-  const ruleMapOutputs = createRuleMap(rulemap.outputs);
   const ruleMapResultOutputs = createRuleMap(rulemap.resultOutputs);
 
-  const [selectedSubmissionInputs, setSelectedSubmissionInputs] = useState<SubmissionData>(ruleMapInputs);
-  const [contextToSimulate, setContextToSimulate] = useState<SubmissionData | null>();
-  const [outputSchema, setOutputSchema] = useState<Record<string, any> | null>(ruleMapOutputs);
```
Comment (timwekkenbc) [inline app/components/SimulationViewer/SimulationViewer.tsx:48]: I noticed the data for this wasn't actually being used anywhere so I removed it and it's subsequent function calls that set the data. Don't know if it was meant to be using for something?

File: app/components/RulesDecisionGraph/LinkRuleComponent.tsx
Comment (brysonjbest) [inline app/components/RulesDecisionGraph/LinkRuleComponent.tsx:1]: Very cool custom component! i like how it handles visually pulling in the rules and setting them up as separate nodes to run the simulation through.

File: app/components/SimulationViewer/SimulationViewer.tsx (hunk 1 above)
Comment (brysonjbest) [reply app/components/SimulationViewer/SimulationViewer.tsx:48]: The things this data was used for are no longer present and have been supplanted. Thank you for removing!

File: package.json [hunk 2]
```diff
@@ -13,12 +13,13 @@
   },
   "dependencies": {
     "@ant-design/nextjs-registry": "^1.0.0",
-    "@gorules/jdm-editor": "^1.0.0",
+    "@gorules/jdm-editor": "^1.9.0",
```
Comment (brysonjbest) [inline package.json:16]: Appreciate the update to the newer version!

File: app/components/SimulationViewer/SimulationViewer.tsx [hunk 3]
```diff
@@ -154,28 +188,41 @@ export default function SimulationViewer({
...
-          contextToSimulate={contextToSimulate}
-          setResultsOfSimulation={setResultsOfSimulation}
-          setOutputsOfSimulation={setOutputSchema}
+          jsonFilename={jsonFile}
+          graphJSON={graphJSON}
+          contextToSimulate={simulationContext}
+          setContextToSimulate={setSimulationContext}
+          simulation={simulation}
+          runSimulation={runSimulation}
+          isEditable={editing}
```
Comment (brysonjbest) [inline app/components/SimulationViewer/SimulationViewer.tsx:211]: Adding the passed down editing logic makes a lot of sense here.

**Review by brysonjbest — APPROVED** [review]
No concerns, looks like a great change that refactors a lot of the current setup, while setting us up for a lot with the addition of linked rules!

**Pull request merged**